src/expr/src/relation/mod.rs

```diff
@@ -1569,6 +1612,7 @@ impl MirRelationExpr {
         let mut first = None;
         let mut second = None;
         let mut rest = None;
+        let mut last = None;
```

Comment:
If we change the output order to be `first`, `rest`, `second` we can avoid the additional variable. Perhaps change uses of `second` to be `last` and then remove `second`, for naming clarity.